src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Read.Span.cs

```diff
@@ -24,7 +24,7 @@ public static partial class JsonSerializer
         /// for <typeparamref name="TValue"/> or its serializable members.
         /// </exception>
         [return: MaybeNull]
-        public static TValue Deserialize<TValue>(ReadOnlySpan<byte> utf8Json, JsonSerializerOptions? options = null)
+        public static TValue Deserialize<[DynamicallyAccessedMembers(MembersAccessedOnRead)] TValue>(ReadOnlySpan<byte> utf8Json, JsonSerializerOptions? options = null)
```

Comment:
@eerhardt I believe that I've asked this already but is there a requirement for us to also update the ref code with these attributes? I know that they only used on the linker fwith the actual implementation, but I'm not sure if we should add them to the refs so that other projects that depend on System.Text.Json throw linker warnings saying they have to add or propagate these attributes as well.

Reply:
Good catch. Yes, we need to add these in the ref assemblies so when we link during the dotnet/runtime build the linker can see the attributes.

Reply:
What about the other PRs where we've added attributes? Y suppose DynamicDependency ones don't really need to be propagated to the refs, but I believe we have added DynamicAccessedMembers attributes on other PRs as well.

Reply:
> DynamicDependency ones don't really need to be propagated to the refs

Correct. Those are explicitly left out.

https://github.com/dotnet/runtime/blob/ecdb7f36e0d97e204194ee23a972160471b44622/eng/DefaultGenApiDocIds.txt#L10

> but I believe we have added DynamicAccessedMembers attributes on other PRs as well.

The ones that @MichalStrehovsky opened have updated the refs. e.g. https://github.com/dotnet/runtime/pull/38331 and https://github.com/dotnet/runtime/pull/38021.

We should update the `ref` assemblies on any place we've added but haven't updated the `ref` files.

Reply:
Fixed, thanks.

Reply:
> We should update the ref assemblies on any place we've added but haven't updated the ref files.

I'll scan through and make a PR to added them, if any.